lib/chewy/type/mapping.rb

```diff
@@ -118,6 +118,35 @@ def field *args, &block
           end
         end
 
+        # Defines an aggregation that can be bound to a query or filter
+        #
+        #   Suppose that a user has posts and each post has ratings
+        #   avg_post_rating is the mean of all ratings
+        #
+        #   class UsersIndex < Chewy::Index
+        #     define_type User do
+        #       field :posts do
+        #         field :rating
+        #       end
+        #
+        #       agg :avg_rating do
+        #         { avg: { field: 'posts.rating' } }
+        #       end
+        #     end
+        #   end
+        def agg *args, &block
+          options = args.extract_options!
+          build_root unless root_object
+
+          @_agg_defs ||= {}
+          @_agg_defs[args.first] = block
```

Comment:
Why don't use class_attribute here to make it work correctly in case of inheritance?